encoders/src/main/java/au/csiro/pathling/encoders/FhirEncoders.java

```diff
@@ -62,19 +65,28 @@
    */
   private final int maxNestingLevel;
 
+  /**
+   * The encoder version to use.
+   */
+  private final int encoderVersion;
+
   /**
    * Consumers should generally use the {@link #forR4()} method, but this is made available for test
    * purposes and additional experimental mappings.
    *
    * @param context the FHIR context to use.
    * @param mappings mappings between Spark and FHIR data types.
    * @param maxNestingLevel maximum nesting level for expansion of recursive data types.
+   * @param encoderVersion the encoder version to use.
    */
   public FhirEncoders(final FhirContext context, final DataTypeMappings mappings,
-      final int maxNestingLevel) {
+      final int maxNestingLevel, int encoderVersion) {
+
+    assert (encoderVersion == 1 || encoderVersion == 2);
```

Comment:
Best not to use `assert` as we don't want to have to rely upon `-ea` being set.

Reply:
OK